doc/design/test-dandisets-1.md

```diff
@@ -0,0 +1,102 @@
+way to (1) discourage the creation and abandonment of exploratory dandisets; (2)
+mark dev dandisets in some way to prevent them from appearing in the main
+listing; and (3) clean up the archive from such abandoned dandisets.
+
+## Proposals
+
+### "Test" flag
+
+This proposes to add a `testing` flag to all dandisets. At creation time, a
+dandiset can optionally have this flag set to prevent it from showing up in
+public listings of all dandisets. These would only show up in the owners' "my
+dandisets" listings (or in all listings shown to the owners). Admins would get a
+checkmark to toggle the display of test dandisets, to help simulate what other
+users are seeing when they visit the archive.
+
+To encourage the use of this feature, the dandiset creation screen would have a
+checkbox along with a description of what this is for ("If you are new to the
+archive and are just exploring, please select this option to avoid this dandiset
+from appearing in public searches.").
+
+### Hide draft-only dandisets
+
+Similarly to hiding explicitly marked test dandisets, we could, by default, also
+hide dandisets that have no published versions (as above, these dandisets would
+still appear for the owners). As the public may *want* to see draft-only
+dandisets, everyone would be able to toggle this mode with a setting.
+
+### Hide dandisets with no assets
+
+Another variation on the previous proposal: hide dandisets that contain no
+assets. These dandisets are a subset of draft-only dandisets, since publishing a
+dandiset requires assets to be present.
+
+This option is problematic on its own--it is a valid use case to declare
+scientific intentions by creating a draft dandiset with no assets, then collect
+and add data only after months or years of real scientific work. That is to say,
+an assetless dandiset that nonetheless has a description, funding agent,
+contact person, owners, etc. may indeed be fully valid.
+
+### Add questions to dandiset creation page
+
+Adding questions such as:
+- Do you welcome collaborators? (Y/N)
+- When do you anticipate posting data to the dandiset?
+
+may help to stem the tide of abandoned dandisets.
+
+The collaborators question, answered positively, would add a badge to the
+dandiset's landing page and search result display indicating the call for
+collaboration.
+
+The anticipation date question would delay considering the dandiset abandoned
+until after that date.
+
+### Enable deletion of dandisets
+
+Part of the problem may be that people simply cannot delete their own abanonded
+dandisets. Offering a delete button with the proper semantics could help here.
+
+The archive does not allow for deletion of *published* dandisets, by design. But
+dandisets that are draft-only should be deleteable.
+
+And, as en extension to that idea, dandisets that are published should be able
+to have their draft dandiset deleted (essentially "archiving" that dandiset, and
+closing it off to further updates).
```

Comment:
would there be a way to reopen this?

Reply:
I like the idea of deleting, but I think we should make it a soft delete, where we truly just de-list the dandiset for a set amount of time (say 2 weeks) before truly deleting it, and the owner would have the ability to get it back. For now this can be done by email, but in the future it would be great to have a recovery process integrated into the UI.

Reply:
If we don't call it "deleting", and instead use something like "frozen" or "finalized" or something, then definitely it would be a reversible operation.

Reply:
See b63eca8f2b3f2dd51484676270a1db25b0b23aae.